src/query/interpreter.cpp

```diff
@@ -2066,6 +2069,10 @@ std::optional<plan::ProfilingStatsWithTotalTime> PullPlan::Pull(AnyStream *strea
   cursor_->Shutdown();
   ctx_.profile_execution_time = execution_time_;
 
+  if (!flags::run_time::GetHopsLimitPartialResults() && (ctx_.hops_limit.has_value() && ctx_.hops_limit.value() < 0)) {
+    throw QueryException("Query exceeded the maximum number of hops.");
+  }
+
```

Comment:
I think it's better maybe to make an object in the context with the value that's maximum, and the current value, so we can ask the object if the hops limit is in place and if the hops limit is exhausted, rather than modifying this limit in place